packages/go_router/README.md

```diff
@@ -134,6 +134,40 @@ methods:
 onTap: () => context.go('/page2')
 ```
 
+## Nested Navigation
+
+The `ShellRoute` route type provdes a way to wrap all sub-routes with a UI shell.
+Under the hood, GoRouter places a Navigator in the widget tree, which is used
+to display matching sub-routes:
+
+```dart
+final  _router = GoRouter(
+  routes: [
+    ShellRoute(
+      builder: (context, state, child) {
+        return AppScaffold(child: child);
+      },
+      routes: <RouteBase>[
+        GoRoute(
+          path: '/albums',
+          builder: (context, state) {
+            return HomeScreen();
+          },
+          routes: <RouteBase>[
+            /// The details screen to display stacked on the inner Navigator.
+            GoRoute(
+              path: 'song/:songId',
+              builder: (BuildContext context, GoRouterState state) {
+                return const DetailsScreen(label: 'A');
+              },
+            ),
+          ],
+        ),
+```
```

Comment:
I'd make sure to close brackets/parenthesis so that folks can copy this code snippet.